src/connectors/curve.mjs

```diff
@@ -100,132 +101,245 @@ function getHorizontalSourceDirection(linkView, route, options) {
     const { sourceBBox } = linkView;
 
     let sourceSide;
+    let rotation;
     if (!sourceBBox.width || !sourceBBox.height) {
         if (sourceBBox.x > route[1].x)
             sourceSide = 'right';
         else
             sourceSide = 'left';
     } else {
-        sourceSide = sourceBBox.sideNearestToPoint(route[0]);
+        if (options.rotate) {
```

Comment:
Can we perhaps execute this branch only if `options.rotate === true` and `linkView.sourceView.model.angle() !== 0`?

The elements being not rotated is the most common case and we should optimize the code for this.